pkg/logs/input/docker/tailer.go

```diff
@@ -67,6 +69,8 @@ func NewTailer(cli *client.Client, containerID string, source *config.LogSource,
 		erroredContainerID: erroredContainerID,
 		reader:             newSafeReader(),
 	}
+	t.readFunc = t.read
```

Comment:
I am not sure to understand why we need this read func, why not injecting a reader provider ?

Reply:
it allows us to mock the read() func